From 596e4e8eb6528b49ce3bf4c52b288d30a388789a Mon Sep 17 00:00:00 2001
From: Olliver Schinagl <oliver@schinagl.nl>
Date: Wed, 7 Aug 2024 16:09:49 +0200
Subject: [PATCH 2/2] examples: Add preinclude and drop special treatment

Instead of manually patching preinclude into the workspace opts, patch
the projectspecs instead.

Note that downstream needs to properly a znp.syscfg with proper pinouts.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>
---
 cc13xx_cc26xx_sdk                                               | 2 +-
 .../ticlang/znp_CC1352P_2_LAUNCHXL_tirtos7_ticlang.projectspec  | 2 ++
 .../ticlang/znp_CC26X2R1_LAUNCHXL_tirtos7_ticlang.projectspec   | 2 ++
 .../ticlang/znp_LP_CC1352P7_4_tirtos7_ticlang.projectspec       | 2 ++
 .../tirtos7/ticlang/znp_LP_CC2652R7_tirtos7_ticlang.projectspec | 2 ++
 .../tirtos7/ticlang/znp_LP_CC2652RB_tirtos7_ticlang.projectspec | 2 ++
 6 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/cc13xx_cc26xx_sdk b/cc13xx_cc26xx_sdk
index 3db978d..08b8b30 160000
--- a/cc13xx_cc26xx_sdk
+++ b/cc13xx_cc26xx_sdk
@@ -1 +1 @@
-Subproject commit 3db978d582bc056fa915a91c65af279bb406924c
+Subproject commit 08b8b30923b9629d5bc3994828ccd451b791520c
diff --git a/examples/rtos/CC1352P_2_LAUNCHXL/zstack/znp/tirtos7/ticlang/znp_CC1352P_2_LAUNCHXL_tirtos7_ticlang.projectspec b/examples/rtos/CC1352P_2_LAUNCHXL/zstack/znp/tirtos7/ticlang/znp_CC1352P_2_LAUNCHXL_tirtos7_ticlang.projectspec
index 303e7cd..b8ac19a 100644
--- a/examples/rtos/CC1352P_2_LAUNCHXL/zstack/znp/tirtos7/ticlang/znp_CC1352P_2_LAUNCHXL_tirtos7_ticlang.projectspec
+++ b/examples/rtos/CC1352P_2_LAUNCHXL/zstack/znp/tirtos7/ticlang/znp_CC1352P_2_LAUNCHXL_tirtos7_ticlang.projectspec
@@ -185,6 +185,8 @@
         </file>
         <file path="../../README.html" openOnCreation="true" excludeFromBuild="true" action="copy">
         </file>
+        <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/stack/config/preinclude.h" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Stack/Config">
+        </file>
         <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/startup/main.c" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Application/StartUp">
         </file>
         <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/common/cui/cui.c" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Application/ui">
diff --git a/examples/rtos/CC26X2R1_LAUNCHXL/zstack/znp/tirtos7/ticlang/znp_CC26X2R1_LAUNCHXL_tirtos7_ticlang.projectspec b/examples/rtos/CC26X2R1_LAUNCHXL/zstack/znp/tirtos7/ticlang/znp_CC26X2R1_LAUNCHXL_tirtos7_ticlang.projectspec
index cdbb6c1..4e7afde 100644
--- a/examples/rtos/CC26X2R1_LAUNCHXL/zstack/znp/tirtos7/ticlang/znp_CC26X2R1_LAUNCHXL_tirtos7_ticlang.projectspec
+++ b/examples/rtos/CC26X2R1_LAUNCHXL/zstack/znp/tirtos7/ticlang/znp_CC26X2R1_LAUNCHXL_tirtos7_ticlang.projectspec
@@ -182,6 +182,8 @@
         </file>
         <file path="../../README.html" openOnCreation="true" excludeFromBuild="true" action="copy">
         </file>
+        <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/stack/config/preinclude.h" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Stack/Config">
+        </file>
         <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/startup/main.c" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Application/StartUp">
         </file>
         <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/common/cui/cui.c" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Application/ui">
diff --git a/examples/rtos/LP_CC1352P7_4/zstack/znp/tirtos7/ticlang/znp_LP_CC1352P7_4_tirtos7_ticlang.projectspec b/examples/rtos/LP_CC1352P7_4/zstack/znp/tirtos7/ticlang/znp_LP_CC1352P7_4_tirtos7_ticlang.projectspec
index d895eb7..fff3283 100644
--- a/examples/rtos/LP_CC1352P7_4/zstack/znp/tirtos7/ticlang/znp_LP_CC1352P7_4_tirtos7_ticlang.projectspec
+++ b/examples/rtos/LP_CC1352P7_4/zstack/znp/tirtos7/ticlang/znp_LP_CC1352P7_4_tirtos7_ticlang.projectspec
@@ -182,6 +182,8 @@
         </file>
         <file path="../../README.html" openOnCreation="true" excludeFromBuild="true" action="copy">
         </file>
+	<file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/stack/config/preinclude.h" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Stack/Config">
+        </file>
         <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/startup/main.c" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Application/StartUp">
         </file>
         <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/common/cui/cui.c" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Application/ui">
diff --git a/examples/rtos/LP_CC2652R7/zstack/znp/tirtos7/ticlang/znp_LP_CC2652R7_tirtos7_ticlang.projectspec b/examples/rtos/LP_CC2652R7/zstack/znp/tirtos7/ticlang/znp_LP_CC2652R7_tirtos7_ticlang.projectspec
index a85890a..2937808 100644
--- a/examples/rtos/LP_CC2652R7/zstack/znp/tirtos7/ticlang/znp_LP_CC2652R7_tirtos7_ticlang.projectspec
+++ b/examples/rtos/LP_CC2652R7/zstack/znp/tirtos7/ticlang/znp_LP_CC2652R7_tirtos7_ticlang.projectspec
@@ -182,6 +182,8 @@
         </file>
         <file path="../../README.html" openOnCreation="true" excludeFromBuild="true" action="copy">
         </file>
+	<file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/stack/config/preinclude.h" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Stack/Config">
+        </file>
         <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/startup/main.c" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Application/StartUp">
         </file>
         <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/common/cui/cui.c" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Application/ui">
diff --git a/examples/rtos/LP_CC2652RB/zstack/znp/tirtos7/ticlang/znp_LP_CC2652RB_tirtos7_ticlang.projectspec b/examples/rtos/LP_CC2652RB/zstack/znp/tirtos7/ticlang/znp_LP_CC2652RB_tirtos7_ticlang.projectspec
index f667ef8..69173d6 100644
--- a/examples/rtos/LP_CC2652RB/zstack/znp/tirtos7/ticlang/znp_LP_CC2652RB_tirtos7_ticlang.projectspec
+++ b/examples/rtos/LP_CC2652RB/zstack/znp/tirtos7/ticlang/znp_LP_CC2652RB_tirtos7_ticlang.projectspec
@@ -181,6 +181,8 @@
         <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/apps/znp/znp_cnf.opts" openOnCreation="false" excludeFromBuild="false" action="copy" targetDirectory="Stack/Config">
         </file>
         <file path="../../README.html" openOnCreation="true" excludeFromBuild="true" action="copy">
+        </file>
+	 <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/stack/config/preinclude.h" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Stack/Config">
         </file>
         <file path="${COM_TI_SIMPLELINK_CC13XX_CC26XX_SDK_INSTALL_DIR}/source/ti/zstack/startup/main.c" openOnCreation="false" excludeFromBuild="false" action="link" targetDirectory="Application/StartUp">
         </file>
-- 
2.46.0

